src/config/rcxml.c

```diff
@@ -241,6 +247,77 @@ fill_region(char *nodename, char *content)
 	}
 }
 
+static void
+fill_action_query(char *nodename, char *content, struct action *action)
+{
+	string_truncate_at_pattern(nodename, ".keybind.keyboard");
+	string_truncate_at_pattern(nodename, ".mousebind.context.mouse");
+
+	if (!strcasecmp(nodename, "query.action")) {
+		current_view_query = NULL;
+	}
+
+	string_truncate_at_pattern(nodename, ".query.action");
+
+	if (!content) {
+		return;
+	}
+
+	if (!current_view_query) {
+		struct wl_list *queries = action_get_querylist(action, "query");
+		if (!queries) {
+			action_arg_add_querylist(action, "query");
+			queries = action_get_querylist(action, "query");
+		}
+		current_view_query = znew(*current_view_query);
+		wl_list_append(queries, &current_view_query->link);
+	}
+
+	if (!strcasecmp(nodename, "identifier")) {
+		current_view_query->identifier = xstrdup(content);
+	} else if (!strcasecmp(nodename, "title")) {
+		current_view_query->title = xstrdup(content);
+	}
+}
+
+static void
+fill_child_action(char *nodename, char *content, struct action *parent,
+	const char *branch_name)
+{
+	string_truncate_at_pattern(nodename, ".keybind.keyboard");
+	string_truncate_at_pattern(nodename, ".mousebind.context.mouse");
+	string_truncate_at_pattern(nodename, ".then.action");
+	string_truncate_at_pattern(nodename, ".else.action");
+
+	if (!strcasecmp(nodename, "action")) {
+		current_child_action = NULL;
+	}
+
+	if (!content) {
+		return;
+	}
+
+	struct wl_list *siblings = action_get_actionlist(parent, branch_name);
+	if (!siblings) {
+		action_arg_add_actionlist(parent, branch_name);
+		siblings = action_get_actionlist(parent, branch_name);
+	}
+
+	if (!strcasecmp(nodename, "name.action")) {
+		if (!strcasecmp(content, "If") || !strcasecmp(content, "ForEach")) {
+			wlr_log(WLR_ERROR, "action '%s' cannot be a child action", content);
+			return;
+		}
+		current_child_action = action_create(content);
+		wl_list_append(siblings, &current_child_action->link);
```

Comment:
Just so we don't forget about this: the return value from `action_create()` can be `NULL` so we need to check for that before calling `wl_list_append()`.

Reply:
Fixed.